### PR TITLE
Google Auth Implemented

### LINKED ISSUE #265: [Backend] Auth Required
Auth to be added
-

A Login, and Sign Up authentication integration are required. 
Auths to be added -

- [ ] Google
- [ ] Apple

### PR DESCRIPTION
**Pull request by PrerakMathur20**
# Description
WIth Firebase Integration, Google Auth has been Implemented for admins only.
The auth is restricted to admin users only to avoid concurrences occurring because of overlapping of roles with same email or usernames.

Fixes #265 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

The auth has been tested with multiple login accounts, with different emails with correct as well as incorrect passwords, and other possible combinations of correct and incorrect details.

Also, include screenshots for the verification and reviewing purpose.

**Login without role as Admin**
 Screenshot 2022-06-24 at 7 31 13 AM

**Login with role as Admin**
Screenshot 2022-06-24 at 7 31 21 AM

**SignUp for Google**
Screenshot 2022-06-24 at 7 31 34 AM

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: React-frontend/src/components/LoginForm.js [hunk 1]
```diff
@@ -54,13 +56,13 @@ const useStyles = makeStyles((theme) => ({
         backgroundColor: theme.palette.primary.extraLight,
         margin: theme.spacing(1.5, 0),
         '&:hover': {
-          backgroundColor: theme.palette.primary.light
+            backgroundColor: theme.palette.primary.light
         }
```
Comment (swapnalshahil) [inline React-frontend/src/components/LoginForm.js:60]: Format the code properly, Avoid extra space or line!

File: React-frontend/src/components/LoginForm.js [hunk 2]
```diff
@@ -173,20 +189,38 @@ function LoginForm({setOpenLogin}) {
                         variant="contained"
                         color="primary"
                         className={classes.submit}
-                        disabled={ auth.isLoading ||!email || !password || !role}
+                        disabled={auth.isLoading || !email || !password || !role}
                         onClick={() => dispatch(login(email, password, role, remember, setOpenLogin))}
                     >
                         {(auth.isLoading) ? (<span>Logging...</span>) : (<span>Login</span>)}
                     </Button>
+
```
Comment (swapnalshahil) [inline React-frontend/src/components/LoginForm.js:197]: Do the same here!

**Review by swapnalshahil — CHANGES_REQUESTED** [review]

File: React-frontend/src/components/LoginForm.js (hunk 1 above)
Comment (PrerakMathur20) [reply React-frontend/src/components/LoginForm.js:60]: Done.

File: React-frontend/src/components/LoginForm.js (hunk 2 above)
Comment (PrerakMathur20) [reply React-frontend/src/components/LoginForm.js:197]: Fixed.

**Review by swapnalshahil — APPROVED** [review]

**Pull request merged**